tests/test_ranges.py

```diff
@@ -1019,6 +1019,48 @@ def test_finite_range(self):
 
 
 class TestFiniteDatetimeRange:
+    @pytest.mark.parametrize(
+        "r1, r2, expected",
+        [
+            [
+                ranges.FiniteDatetimeRange(
+                    start=datetime.datetime(2000, 1, 1),
+                    end=datetime.datetime(2000, 1, 3),
+                ),
+                ranges.FiniteDatetimeRange(
+                    start=datetime.datetime(2000, 1, 2),
+                    end=datetime.datetime(2000, 1, 4),
+                ),
+                True,
+            ],
+            [
+                ranges.FiniteDatetimeRange(
+                    start=datetime.datetime(2000, 1, 2),
+                    end=datetime.datetime(2000, 1, 4),
+                ),
+                ranges.FiniteDatetimeRange(
+                    start=datetime.datetime(2000, 1, 1),
+                    end=datetime.datetime(2000, 1, 3),
+                ),
+                False,
+            ],
+            [
+                ranges.FiniteDatetimeRange(
+                    start=datetime.datetime(2000, 1, 1),
+                    end=datetime.datetime(2000, 1, 3),
+                ),
+                ranges.FiniteDatetimeRange(
+                    start=datetime.datetime(2000, 1, 1),
+                    end=datetime.datetime(2000, 1, 5),
+                ),
+                # False, since only `start` is considered.
```

Comment:
This is consistent with the base class implementation - `end` is not considered.